src/reflect/scala/reflect/internal/Names.scala

```diff
@@ -27,7 +27,7 @@ trait Names extends api.Names {
   // detect performance regressions.
   //
   // Discussion: https://groups.google.com/forum/#!search/biased$20scala-internals/scala-internals/0cYB7SkJ-nM/47MLhsgw8jwJ
-  protected def synchronizeNames: Boolean = false
+  protected def synchronizeNames: Boolean = true
```

Comment:
do we need this?

Reply:
Yes we do.
Umad reports violation in this class:

```
[WARN] Method accessed from multiple threads (main, scalac-parser-worker-1): scala.reflect.internal.SymbolTable.scala$reflect$internal$Names$$nc_$eq(int)
    scala.reflect.internal.SymbolTable.scala$reflect$internal$Names$$nc_$eq(SymbolTable.scala)
    scala.reflect.internal.Names.enterChars(Names.scala:78)
    scala.reflect.internal.Names.body$1(Names.scala:116)
    scala.reflect.internal.Names.newTermName(Names.scala:127)
    scala.reflect.internal.Names.newTermName$(Names.scala:96)
    scala.reflect.internal.SymbolTable.newTermName(SymbolTable.scala:18)
    scala.reflect.internal.Names.newTermName(Names.scala:83)
    scala.reflect.internal.Names.newTermName$(Names.scala:82)
    scala.reflect.internal.SymbolTable.newTermName(SymbolTable.scala:18)
    scala.reflect.internal.Names.newTermName(Names.scala:85)
```

What `synchronizeNames` do is enabling synchronization on the `newTermName`.
It looks like something designed exactly for our use case so I happily used it.
There were some concerns regarding enabling it globally though (mostly performance ones): https://groups.google.com/forum/#!search/biased$20scala-internals/scala-internals/0cYB7SkJ-nM/47MLhsgw8jwJ